modules/programs/fish.nix

```diff
@@ -343,9 +343,21 @@ in {
         # if we haven't sourced the general config, do it
         if not set -q __fish_general_config_sourced
 
-          set -p fish_function_path ${pkgs.fish-foreign-env}/share/fish-foreign-env/functions
-          fenv source ${config.home.profileDirectory}/etc/profile.d/hm-session-vars.sh > /dev/null
-          set -e fish_function_path[1]
+          ${
+            builtins.concatStringsSep "\n" (attrsets.mapAttrsToList
+              (envVar: value: "set -xg ${envVar} ${value}")
+              config.home.sessionVariables)
+          }
+
+          set -l p "$HOME/.nix-profile/bin"
+          if not contains $p $PATH
+            set -p PATH $p
+          end
+
+          set -l p "$HOME/.nix-defexpr/channels"
+          if not contains $p $NIX_PATH
+            set -xgp NIX_PATH $p
+          end
```

Comment:
I think this belongs in `targets/generic-linux.nix` (and perhaps `targets/darwin.nix`). But definitely need to use `${config.home.profileDirectory}` instead of `$HOME/.nix-profile`.